client/components/FormBuilder.jsx

```diff
@@ -0,0 +1,70 @@
+/* eslint react/prop-types: 0 */
+/* eslint react/destructuring-assignment: 0 */
+/* eslint react/no-array-index-key: 0 */
+
+import React from 'react';
+import {
+  Avatar, Button, Container, Box, Typography, CssBaseline, Grid, TextField,
+} from '@material-ui/core';
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import { Alert } from '@material-ui/lab';
+import { useStyles } from '../styles/useStyles';
+import { Copyright } from './Copyright';
+
+export function FormBuilder(props) {
+  const classes = useStyles();
+
+  return (
+    <Container component="main" maxWidth="xs">
+      <CssBaseline />
+      <div className={classes.paper}>
+        <Avatar className={classes.avatar}>
+          <AccountCircleIcon />
+        </Avatar>
+        <Typography component="h1" variant="h5">
+          {props.header}
+        </Typography>
+        <form className={classes.form} onSubmit={props.onSubmit}>
+          {/* {console.log(props.fields)} */}
+          <Grid container spacing={2}>
+            {props.fields.map((field, index) => (
+              <Grid item xs={12}>
+                <TextField
+                  variant="outlined"
+                  required
+                  fullWidth
+                  key={index}
+                  name={field.name}
+                  label={field.label}
+                  type={field.type}
+                  id={field.id}
+                  autoComplete={field.autoComplete}
+                  value={field.value}
```

Comment:
I tested it without `value={field.value}` and it worked.

Reply:
It will work without the value field, but it won't be a controlled component without it. Essentially the React docs recommend keeping components controlled so that at all point the value is derived from state rather than the DOM element.